monitoring: use histograms for api latency and cycle time metrics

alexbrand opened the pull request:
Fixes #163 

- Changed the API latency and OpenStack cycle time metrics to use a histogram instead of a gauge. 
- Updated the dashboards to display 50th,75th, 90th, and 100th percentiles.
- Updated metrics documentation

![image](https://user-images.githubusercontent.com/545723/41795686-a38e7670-7631-11e8-8e49-32feacc03700.png)

I suspect we'll have to make the bucket sizes configurable, but I think we can follow up on that when time comes.

stevesloka commented:
I don't quite understand how the histograms are implemented in this visualization. Typically I see histograms like this: http://docs.grafana.org/features/panels/heatmap/#histograms-and-buckets

What does the `current` metric mean for each bucket? If we had multiple discoverers, then they you would get 4 more data series?

alexbrand commented:
This visualization shows the 50, 75, 90 and max percentiles for each discoverer. It was confusing for me at first, but after reading a bit I now understand it better. The way I read the graphs are:

"50% (or 75%, or 90%) of the requests are taking less than X milliseconds".

If we have multiple discoverers, then we would indeed have 4 more data series on the graph.

I'll let @rosskukulinski weigh in as well as this is somewhat new for me still.

rosskukulinski commented:
> "50% (or 75%, or 90%) of the requests are taking less than X milliseconds".

@alexbrand is correct about what's being visualized

That said, @stevesloka - you're right that these are not _actually_ histogram visualizations like the ones you linked to.  This is partly because it's only in the latest release of Grafana (#148) is there a proper Prometheus histograph panel.  It's also a common pattern to display the latency percentiles (especially 

@alexbrand recommend:

* changing legend for each graph to a table underneath the graph and display the 'avg' and 'current' values.
* Recommend only showing 50th and 90th percentiles by default to minimize clutter when there are many clusters.

stevesloka commented:
What do the `current` values in the legend mean? Also, Why does it show 100% as a bucket?

@rosskukulinski Should we update the Grafana version and use the proper histogram panel?

rosskukulinski commented:
100% is basically Inf -- all values are less than this number.  Its the same thing as maximum latency value recorded.

`current` is the most recent calculated value.  This is as opposed to `max` which is the max value for the rolling window Grafana is currently displaying, `min`, which is the min value for the rolling window, and `avg` which is the average over the rolling window.

I do think we should update Grafana, but I do not think it's necessary for this PR.  FWIW, I prefer line-graph representation of latencies because you can display many different lines.  Doing that with true histograms is more complex and confusing to understand.

alexbrand commented:
Updated with 50th and 99th percentile, and also updated legend to show up as table with avg and current values.

![screen shot 2018-06-25 at 4 43 12 pm](https://user-images.githubusercontent.com/545723/41874794-e4753e96-7896-11e8-9717-fa68f3bd0a74.png)

rosskukulinski commented:
I don't have an OS system up to test with, but latest attached image LGTM.

stevesloka commented:
LGTM, @alexbrand could you rebase to update?

alexbrand commented:
Rebased. Thanks @stevesloka!
